src/OnlineSales/Entities/Deal.cs

```diff
@@ -0,0 +1,57 @@
+﻿// <copyright file="Deal.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text.Json.Serialization;
+using OnlineSales.DataAnnotations;
+
+namespace OnlineSales.Entities;
+
+[Table("deal")]
+[SupportsElastic]
+[SupportsChangeLog]
+public class Deal : BaseEntity
+{
+    public int AccountId { get; set; }
+
+    [JsonIgnore]
+    [ForeignKey("AccountId")]
+    public virtual Account? Account { get; set; }
+
+    public int DealPipelineId { get; set; }
+
+    [JsonIgnore]
+    [ForeignKey("DealPipelineId")]
+    public virtual DealPipeline? DealPipeline { get; set; }
+
+    public int PipelineStageId { get; set; }
+
+    [JsonIgnore]
+    [ForeignKey("PipelineStageId")]
+    public virtual PipelineStage? PipelineStage { get; set; }
```

Comment:
I think it will be be better if 'stage' itself will contain information about it's current status, not deal.